core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/StopStartDownscaleGetRecoveryCandidatesHandler.java

```diff
@@ -0,0 +1,138 @@
+import com.sequenceiq.cloudbreak.converter.spi.InstanceMetaDataToCloudInstanceConverter;
+import com.sequenceiq.cloudbreak.core.flow2.stack.CloudbreakFlowMessageService;
+import com.sequenceiq.cloudbreak.cloud.event.instance.StopStartDownscaleGetRecoveryCandidatesRequest;
+import com.sequenceiq.cloudbreak.cloud.event.instance.StopStartDownscaleGetRecoveryCandidatesResult;
+import com.sequenceiq.cloudbreak.dto.StackDto;
+import com.sequenceiq.cloudbreak.service.stack.StackDtoService;
+import com.sequenceiq.cloudbreak.service.stopstart.RecoveryCandidateCollectionService;
+import com.sequenceiq.cloudbreak.view.InstanceMetadataView;
+
+import reactor.bus.Event;
+import reactor.bus.EventBus;
+
+@Component
+public class StopStartDownscaleGetRecoveryCandidatesHandler implements CloudPlatformEventHandler<StopStartDownscaleGetRecoveryCandidatesRequest> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StopStartDownscaleGetRecoveryCandidatesHandler.class);
+
+    @Inject
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Inject
+    private CloudbreakFlowMessageService flowMessageService;
+
+    @Inject
+    private StackDtoService stackDtoService;
+
+    @Inject
+    private RecoveryCandidateCollectionService recoveryCandidateCollectionService;
+
+    @Inject
+    private InstanceMetaDataToCloudInstanceConverter instanceMetaDataToCloudInstanceConverter;
+
+    @Inject
+    private EventBus eventBus;
+
+    @Override
+    public Class<StopStartDownscaleGetRecoveryCandidatesRequest> type() {
+        return StopStartDownscaleGetRecoveryCandidatesRequest.class;
+    }
+
+    @Override
+    public void accept(Event<StopStartDownscaleGetRecoveryCandidatesRequest> event) {
+        StopStartDownscaleGetRecoveryCandidatesRequest request = event.getData();
+        LOGGER.info("StopStartDownscaleGetRunningInstancesHandler: {}, with request: {}", request.getResourceId(), request);
+
+        if (!request.isFailureRecoveryEnabled()) {
+            StopStartDownscaleGetRecoveryCandidatesResult result = new StopStartDownscaleGetRecoveryCandidatesResult(request.getResourceId(),
+                    Collections.emptyList(), request.getHostGroupName(), request.getHostIds());
+            notify(result, event);
+            return;
+        }
+
+        CloudContext cloudContext = request.getCloudContext();
+
+        try {
+            CloudConnector connector  = cloudPlatformConnectors.get(cloudContext.getPlatformVariant());
+            AuthenticatedContext ac = getAuthenticatedContext(request, cloudContext, connector);
+
+            StackDto stack = stackDtoService.getById(request.getResourceId());
+
+            Set<String> startedInstanceIdsOnCloudProvider = collectStartedInstancesFromCloudProvider(connector, ac, request.getAllInstancesInHostGroup())
+                    .stream()
+                    .map(CloudInstance::getInstanceId).collect(Collectors.toSet());
+
```

Comment:
duplicated code, upscale handler has the same code

Reply:
at least a common method needed which can return with `startedInstanceIdsOnCloudProvider` in both handler

Reply:
Addressed with an additional common method and added corresponding UT

Reply:
thank you!